cmd/snap-confine/snap-confine.c

```diff
@@ -515,16 +515,47 @@ int main(int argc, char **argv)
 	return 1;
 }
 
-static void enter_classic_execution_environment(void)
+static void enter_classic_execution_environment(sc_invocation * inv)
 {
 	/* 'classic confinement' is designed to run without the sandbox inside the
 	 * shared namespace. Specifically:
-	 * - snap-confine skips using the snap-specific mount namespace
+	 * - snap-confine skips using the snap-specific mount namespace (unless a
+	 parallel installed classic snap)
 	 * - snap-confine skips using device cgroups
 	 * - snapd sets up a lenient AppArmor profile for snap-confine to use
 	 * - snapd sets up a lenient seccomp profile for snap-confine to use
 	 */
+
 	debug("skipping sandbox setup, classic confinement in use");
+
+	/* Parallel installed classic snap get special handling */
+	if (!sc_streq(inv->snap_instance, inv->snap_name)) {
```

Comment:
Let's try to be bold and always do this. We may have to back down and revert this but for the purpose of testing this more widely and for the purpose of less edge cases I would like to see how it works if it is always done.
If we cannot use it for whatever reason we will know sooner.

Reply:
if I'm not mistaken, always having classic snaps always enter mount namespaces would be a pre-req for supporting content interface sharing with classic snaps, correct? I am +1 on more testing anyways

Reply:
Yes but the reason I want to do this here is to see it break, if it does, without having to also be in an edge case (classic instance)

Reply:
Ping about this. I mainly want to do this to evaluate the full impact of the change across the test suite.

Reply:
Be bold, but behind the experimental feature flag, right?

Reply:
Yes, exactly

Reply:
The code is behind a feature flag now.

Reply:
We are not bold. Can we please be bold here?